device/cluster.cpp

```diff
@@ -578,43 +604,43 @@ Cluster::Cluster(
 }
 
 Cluster::Cluster(
-    const std::string& sdesc_path,
-    const std::set<chip_id_t>& target_devices,
+    std::unordered_map<chip_id_t, std::unique_ptr<Chip>>& chips,
```

Comment:
@pjanevskiTT I couldn't resist offering this constructor, once I figured out that it enables me to do the create_mock_cluster thing without the hacked private constructor.
The static create_mock_cluster function can now be viewed just as a helper function to call the constructor with the right arguments for you.
I intentionally left out this constructor from the api tests, because that should not be a recommended way to do things for our users (but I still like that we leave that flexibility available).

Reply:
> I intentionally left out this constructor from the api tests, because that should not be a recommended way to do things for our users (but I still like that we leave that flexibility available).

Reply:
I'd like to keep more flexibility until we have a better grasp over what do our customers needs. I envision this might unblock some experimentation without our intevention.

Reply:
agree